### PR TITLE
init bdd

### PR DESCRIPTION
**Pull request by JorisPineau**
## Description

Please include a summary of the change or which issue is fixed.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [ ] Frontend
- [x] Backend
- [ ] Other: .....

## General checklist:

- [ ] My code follows the style guidelines of this project
- [x] I ran `bun run format`/`bun run lint` before commit
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation where needed
- [ ] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [ ] I have updated `develop` and merged to my branch before submitting pull request
- [ ] My pull request generate no conflicts with `develop` branch
- [x] I requested code review from other team members

## Frontend checklist:

- [ ] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [ ] My `Javascript` generate no new console errors
- [ ] I tested my code cross browsers
- [ ] My slice is pixel perfect for both desktop and mobile according to design
- [ ] I conducted basic QA to assure all features are working
- [ ] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [ ] I tested admin by manually adding content from zero
- [ ] I created easy to use admin experience which is self-explanatory
- [ ] I added description to admin fields in hard-to-understand areas
- [x] I followed guidelines for naming `Typescript` variables
- [ ] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by Abecker37 — DISMISSED** [review]

**Review by herve-perel — APPROVED** [review]

**Review by Abecker37 — APPROVED** [review]
Bien la résolution des conflits

**Review by herve-perel — APPROVED** [review]

File: packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  // Migration code that reverts the database to the previous state.
+  await db.schema.dropTable('user').ifExists().execute();
+  await db.schema.dropTable('trip').ifExists().execute();
+  await db.schema.dropTable('car').ifExists().execute();
+  await db.schema.dropTable('car_type').ifExists().execute();
+  await db.schema.dropTable('reservation').ifExists().execute();
+  await db.schema.dropTable('seat_comparison').ifExists().execute();
+  await db.schema.dropTable('checkpoint_trip').ifExists().execute();
+  await db.schema.dropTable('notice').ifExists().execute();
+  await db.schema.dropTable('messaging').ifExists().execute();
```
Comment (YDrogen) [inline packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts:170]: Il faut inverser
Les tables créées en dernier doivent être supprimées en premier

File: packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  await db.schema
+    .createTable('user')
+    .addColumn('id', 'int8', (col) =>
+      col.autoIncrement().primaryKey().unsigned(),
+    )
+    .addColumn('firstname', 'varchar(100)', (col) => col.notNull())
+    .addColumn('lastname', 'varchar(100)', (col) => col.notNull())
+    .addColumn('email', 'varchar(100)', (col) => col.notNull().unique())
+    .addColumn('password', 'varchar(255)', (col) => col.notNull())
+    .addColumn('birthday', 'varchar(10)', (col) => col.notNull())
```
Comment (YDrogen) [inline packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts:18]: C'est pas censé être une date ?
\+ `birthdate`

File: packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  await db.schema
+    .createTable('trip')
+    .addColumn('id', 'int8', (col) =>
+      col.autoIncrement().primaryKey().unsigned(),
+    )
+    .addColumn('driver_id', 'int8', (col) =>
+      col.references('driver.id').notNull().unsigned(),
+    )
+    .addColumn('created_at', 'datetime', (col) => col.notNull())
+    .addColumn('date', 'varchar(10)', (col) => col.notNull())
```
Comment (YDrogen) [inline packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts:40]: idem ?

File: packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    .addColumn('kilometer', 'decimal(6, 1)', (col) => col.notNull().unsigned())
+    .addColumn('travel_time', 'varchar(5)', (col) => col.notNull())
```
Comment (YDrogen) [inline packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts:42]: "temps" de trajet en chaîne de caractères ?

File: packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+    .addColumn('seat_available', 'int2', (col) => col.notNull().unsigned())
+    .addColumn('price', 'decimal(6, 2)', (col) => col.notNull().unsigned())
```
Comment (YDrogen) [inline packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts:44]: On a dit que le prix ne devait pas avoir de virgules !

File: packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    .addColumn('auto_validation', 'boolean', (col) => col.notNull())
+    .addColumn('comment', 'text')
+    .addColumn('baby_allowed', 'boolean', (col) => col.notNull())
+    .addColumn('non_vaccinated_allowed', 'boolean', (col) => col.notNull())
+    .addColumn('animal_allowed', 'boolean', (col) => col.notNull())
+    .addColumn('smoker_allowed', 'boolean', (col) => col.notNull())
+    .addColumn('with_toll', 'boolean', (col) => col.notNull())
```
Comment (YDrogen) [inline packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts:51]: les booléens doivent commencer par
`is` `has` `should` `can` `did` `will`

File: packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+    .addColumn('user_id', 'int8', (col) =>
+      col.references('user.id').notNull().unsigned(),
+    )
+    .addColumn('number_seat', 'int2', (col) => col.notNull().unsigned())
+    .addColumn('validated_at', 'datetime', (col) => col.notNull())
+    .addColumn('created_at', 'datetime', (col) => col.notNull())
+    .execute();
+
+  await db.schema
+    .createTable('seat_comparison')
```
Comment (YDrogen) [inline packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts:97]: Faudrait renommer cette table

File: packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+    )
+    .execute();
+
+  await db.schema
+    .createTable('messaging')
+    .addColumn('id', 'int8', (col) =>
+      col.autoIncrement().primaryKey().unsigned(),
+    )
+    .addColumn('comment', 'text', (col) => col.notNull())
+    .addColumn('date', 'datetime', (col) => col.notNull())
```
Comment (YDrogen) [inline packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts:149]: Tu peux la nommer `sent_at` pour garder une constance avec les autres tables

File: packages/types/src/database.ts
Comment (YDrogen) [inline packages/types/src/database.ts:1]: Répercuter les modifs ici

**Review by YDrogen — CHANGES_REQUESTED** [review]

**Pull request closed without merging**